tangelo/problem_decomposition/dmet/dmet_problem_decomposition.py

```diff
@@ -179,6 +185,10 @@ def __init__(self, opt_dict):
         # If save_results in _oneshot_loop is True, the dict is populated.
         self.solver_fragment_dict = dict()
 
+        # To keep track the number of iteration (was done with an energy list before).
+        # TODO: A decorator function to do the same thing?
```

Comment:
That sounds overkill. Beyond the exercise: do you think it is the best use of your time, or does it genuinely add value to the codebase ? If not, then forget about it.